sdk/servicebus/service-bus/src/serializers/subscriptionResourceSerializer.ts

```diff
@@ -105,7 +106,12 @@ export function buildSubscription(rawSubscription: any): SubscriptionProperties
 export function buildSubscriptionRuntimeProperties(
   rawSubscription: any
 ): SubscriptionRuntimeProperties {
-  const messageCountDetails = getMessageCountDetails(rawSubscription[Constants.COUNT_DETAILS]);
+  type MakeTOptional<M, T extends keyof M> = Partial<Pick<M, T>> & Omit<M, T>;
+  const messageCountDetails: MakeTOptional<
+    MessageCountDetails,
+    "scheduledMessageCount"
+  > = getMessageCountDetails(rawSubscription[Constants.COUNT_DETAILS]);
```

Comment:
but why...

I tried the below and it builds just fine

```js
  const messageCountDetails = getMessageCountDetails(rawSubscription[Constants.COUNT_DETAILS]);
  delete messageCountDetails.scheduledMessageCount;
```

Reply:
Wait what... How could you?
![image](https://user-images.githubusercontent.com/10452642/88413746-9232aa00-cd90-11ea-85af-370bbb3389b0.png)

Reply:
Interesting, the build succeeded for me but the squiggly lines are present saying that "The operand of a 'delete' operator must be optional."

Reply:
What version of TypeScript is your vscode using? (Should show in the lower right-hand side) Maybe it's a newer one than what the project compiles with.

Reply:
So, TS had a change https://github.com/microsoft/TypeScript/pull/37921 to add this check, they reverted it in https://github.com/microsoft/TypeScript/pull/38154 in TS@3.9 since it broke VSCode(https://github.com/microsoft/vscode/issues/96022) and planned this check for a later release TS@4.0 and hence reverted the revert in https://github.com/microsoft/TypeScript/pull/38173 for TS@4.0.

Technically, we should not be able to delete the non-optional properties moving forward.

Reply:
> What version of TypeScript is your vscode using? (Should show in the lower right-hand side) Maybe it's a newer one than what the project compiles with.

My VS Code is on `typescript@next` - Currently, I'm on this version - https://www.npmjs.com/package/typescript/v/4.0.0-dev.20200717

Reply:
I am on 3.9.6

Even if we want to future proof ourself, I'd much rather get rid of `getMessageCountDetails()` and inline that code rather than use the above in order to help readability of code.

Reply:
https://github.com/Azure/azure-sdk-for-js/pull/10250/commits/8cb0478023d1577edc72bc009fac27e6e6c3bb12, fine?